### PR TITLE
Enhance builds and push

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Added a way to build and push images automatically based on env variables.

## How Has This Been Tested?

1. Change `.env.local` with values to your own private repo
2. Run `make build`
3. Run `make push`
4. Run `make kind-deployment`

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [X] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work.
- [X] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/README.md [hunk 1]
```diff
@@ -46,3 +46,68 @@ make kind-deployment
 
 You can find the OpenAPI specification for the Model Registry UI in the [openapi](./api/openapi) directory.
 A live version of the OpenAPI specification can be found [here](https://editor.swagger.io/?url=https://raw.githubusercontent.com/kubeflow/model-registry/main/clients/ui/api/openapi/mod-arch.yaml).
+
+## Environment Variables
+
+The following environment variables are used to configure the deployment and development environment for the Model Registry project. These variables should be defined in a `.env.local` file in the root directory of the project. **This values will affect the build and push commands**.
```
Comment (Griffin-Sullivan) [inline clients/ui/README.md:52]:
```suggestion
The following environment variables are used to configure the deployment and development environment for the Model Registry UI. These variables should be defined in a `.env.local` file in the `clients/ui` directory of the project. **This values will affect the build and push commands**.
```

File: clients/ui/manifests/base/kustomization.yaml [hunk 2]
```diff
@@ -2,15 +2,18 @@ apiVersion: kustomize.config.k8s.io/v1beta1
...
-    newName: kubeflow/model-registry-ui:latest
-  - name: model-registry-bff-image
-    newName: kubeflow/model-registry-bff:latest
+- name: model-registry-bff-image
+  newName: kubeflow/model-registry-bff
+  newTag: latest
+- name: model-registry-ui-image
+  newName: kubeflow/model-registry-ui
+  newTag: test-eder
+namespace: latest
```
Comment (Griffin-Sullivan) [inline clients/ui/manifests/base/kustomization.yaml:19]: Could you give some context for these changes?

Comment (lucferbux) [reply clients/ui/manifests/base/kustomization.yaml]: Yeah, that's a typo, already fixed, thanks @Griffin-Sullivan

**Comment by alexcreasy** [comment]
Works for me, thanks @lucferbux lgtm

**Review by alexcreasy — APPROVED** [review]

**Pull request merged**